tests/devapps/Intune-xamarin-Android/MAMWEAuthCallback.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Runtime;
+using Android.Views;
+using Android.Widget;
+using Microsoft.Intune.Mam.Policy;
+
+namespace Intune_xamarin_Android
+{
+    /// <summary>
+    /// Required by the MAM SDK. A token may be needed very early in the app lifecycle so the ideal
+    /// place to register the callback is in the OnMAMCreate() method of the app's implementation
+    /// of IMAMApplication.
+    /// See https://docs.microsoft.com/en-us/intune/app-sdk-android#account-authentication
+    /// </summary>
+    class MAMWEAuthCallback : Java.Lang.Object, IMAMServiceAuthenticationCallback

```

Comment:
What is this class needed for? (Skimmed the docs but isn't clear.) When MAM tries to enroll the app, it needs this token? What if the silent auth fails, no interactive call?

Reply:
This is a callback when ```IMAMEnrollmentManager``` starts enrollment. ```AcquireToken``` is passed url of the MAM service as resource and expects AT from the app for it. The silent will not fail as it is called after user does interactive login. So user has been authenticated at this time. It may be confusing that silent is called after ATI. But the call is to get AT for MAM service, (which is different from the original resource that the user requested in ATI.)